spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/support/DatabaseType.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.task.repository.support;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.sql.DataSource;
+
+import org.springframework.jdbc.support.JdbcUtils;
+import org.springframework.jdbc.support.MetaDataAccessException;
+import org.springframework.util.StringUtils;
+
+
+/**
+ * Enum representing a database type, such as DB2 or oracle.  The type also
+ * contains a product name, which is expected to be the same as the product name
+ * provided by the database driver's metadata.
+ *
+ * @author Glenn Renfro
+ */
+public enum DatabaseType {
+
+	DERBY("Apache Derby"),
+	DB2("DB2"),
+	DB2ZOS("DB2ZOS"),
+	HSQL("HSQL Database Engine"),
+	SQLSERVER("Microsoft SQL Server"),
+	MYSQL("MySQL"),
+	ORACLE("Oracle"),
+	POSTGRES("PostgreSQL"),
+	SYBASE("Sybase"),
+	H2("H2"),
+	SQLITE("SQLite");
```

Comment:
We have all these values, yet we're only providing the HSQLDB schema...I'd only add the values if we have the schemas for them.